codespell_lib/_codespell.py

```diff
@@ -887,10 +891,10 @@ def parse_file(
     bad_count = 0
     lines = None
     changed = False
+    encoding: str | None = "utf-8"
```

Comment:
Perhaps leave these for a different PR?

Reply:
I can't. I must add the type definition, otherwise mypy complains. But if I add it, then  ruff complains about too many statements in this function (120) - we're at exactly 119. And so I have to remove the encoding = "utf-8" from the filename if.

Do you prefer to do a noqa for the too many statements? not a problem for me - but we do need the types.

Reply:
In that case, please keep the new types required to keep MyPy happy.